design/use_cases/indoor_localization_use_case.md

```diff
@@ -0,0 +1,15 @@
+# Indoor Localization
+As a Robot user I want my robot to know its location on a given map of an indoor area so that it can move around the area
+
+## More details
+- Why is this needed?
+   - This is needed for indoor robot navigation in most (all?) cases
+   - Example: a courier robot in a logistics warehouse
+
+- What is the expected user interaction?
+   - The user should be able to specify a map to use and a location on that map for the robot
+   - The robot should be able to deduce it's own position on a map autonomously
+
+- Are there any non-functional requirements? (build system, tools, performance, etc)
```

Comment:
A tool should provide feedback about the overall confidence of the localization. Also in a non-expert fashion (Localized, Uncertain, Lost ...).

Reply:
The same manner as ROS1 `move_base`, ROS2 localization could offer configurable/customizable recovery-behavior.